modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -381,7 +381,7 @@ class Session implements ISession {
         this.dag = new DAG()
 
         // -- init output dir
-        this.outputDir = FileHelper.toCanonicalPath(config.outputDir ?: 'results')
+        this.outputDir = FileHelper.toCanonicalPath(config.outputDir ?: config.navigate('params.outdir') ?: 'results')
```

Comment:
Let's keep this PR focused on the channel factories, we can address publishDir support separately

Reply:
removed